_rules/aria-required-context-role-ff89c9.md

```diff
@@ -0,0 +1,229 @@
+
+## Assumptions
+
+If the [explicit semantic role](#explicit-role) on the target element is incorrectly used, and any relationships between elements are already programmatically determinable, failing this rule may not result in accessibility issues for users of assistive technologies, and it should then not be considered a failure under WCAG success criterion 1.3.1 Info and Relationships.
+
+## Accessibility Support
+
+- User agents do not all have the same accessibility tree. Particularly the method of deriving which element owns which other elements varies between browsers. This can lead to different results from this rule, depending on which accessibility tree is used as input.
+- `aria-owns` has limited support in some user agents.
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [Required Context Role](https://www.w3.org/TR/wai-aria-1.1/#scope)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Element with [explicit semantic role](#explicit-role) `listitem` is contained within its [required context role](https://www.w3.org/TR/wai-aria-1.1/#scope) `list`, expressed as an [explicit semantic role](#explicit-role).
+
+```html
+<div role="list">
+	<div role="listitem">List item 1</div>
+</div>
+```
+
+#### Passed Example 2
+
+Element with [explicit semantic role](#explicit-role) `listitem` is contained within its [required context role](https://www.w3.org/TR/wai-aria-1.1/#scope) `list`, through the [implicit semantic role](#implicit-role) of `ul`.
+
+```html
+<ul>
+	<div role="listitem">List item 1</div>
+</ul>
+```
+
+#### Passed Example 3
+
+Element contained within its [required context role](https://www.w3.org/TR/wai-aria-1.1/#scope) even though it is not a direct child of the [context role](https://www.w3.org/TR/wai-aria-1.1/#scope).
+
+```html
+<div role="list">
+	<div role="presentation">
+		<div role="listitem">List item 1</div>
+	</div>
+</div>
+```
+
+#### Passed Example 4
+
+`aria-owns` used to give the target element the right [context role](https://www.w3.org/TR/wai-aria-1.1/#scope).
+
+```html
+<div role="list" aria-owns="id1"></div>
+<div id="id1" role="listitem">List item 1</div>
+```
+
+#### Passed Example 5
+
+`aria-owns` trumps ownership by closest ancestor, giving the element with [explicit semantic role](#explicit-role) of `listitem` the correct [context role](https://www.w3.org/TR/wai-aria-1.1/#scope).
+
+```html
+<div role="list" aria-owns="id1">
+	<div role="navigation">
+		<div id="id1" role="listitem">List item 1</div>
+	</div>
+</div>
+```
+
+#### Passed Example 6
+
+Since implicit ownership can cross shadow boundaries, the element with the [explicit semantic role](#explicit-role) of `listitem` is contained within its [required context role](https://www.w3.org/TR/wai-aria-1.1/#scope) `list`.
```

Comment:
```suggestion
Since implicit ownership can cross shadow boundaries, the element with the [explicit semantic role](#explicit-role) of `listitem` is contained within its [required context role](https://www.w3.org/TR/wai-aria-1.1/#scope) of `list`.
```